src/protocols/rdp/rdp.c

```diff
@@ -515,6 +526,9 @@ static int guac_rdp_handle_connection(guac_client* client) {
     /* Set default pointer */
     guac_common_cursor_set_pointer(rdp_client->display->cursor);
 
+    rdp_inst->settings->TcpAckTimeout = get_envvar_uint32("GUACD_RDP_TCP_ACK_TIMEOUT", 30000);  // FreeRDP default 9000
```

Comment:
if the default is `9000` (so 9s?) 
How did it manage to fail with a round-trip time of 600 ms ?

Reply:
I'm not sure, I didn't dig any deeper after changing it on a hunch and it resolved the issue reliably.